app/controllers/chargeback_controller.rb

```diff
@@ -597,23 +702,48 @@ def cb_rate_set_form_vars
   def cb_rate_get_form_vars
     @sb[:rate] = @edit[:rate]
     @edit[:new][:description] = params[:description] if params[:description]
-    @edit[:new][:details].each_with_index do |_detail, i|
-      @edit[:new][:details][i][:rate] = params["rate_#{i}".to_sym] if params["rate_#{i}".to_sym]
-      @edit[:new][:details][i][:per_time] = params["per_time_#{i}".to_sym] if params["per_time_#{i}".to_sym]
-      @edit[:new][:details][i][:per_unit] = params["per_unit_#{i}".to_sym] if params["per_unit_#{i}".to_sym]
+    @edit[:new][:details].each_with_index do |_detail, detail_index|
+      @edit[:new][:details][detail_index][:per_time] =
+        params["per_time_#{detail_index}".to_sym] if params["per_time_#{detail_index}".to_sym]
+      @edit[:new][:details][detail_index][:per_unit] =
+        params["per_unit_#{detail_index}".to_sym] if params["per_unit_#{detail_index}".to_sym]
       # Add currencies to chargeback_controller.rb
-      @edit[:new][:details][i][:currency] = params[:currency] if params[:currency]
+      @edit[:new][:details][detail_index][:currency] = params[:currency] if params[:currency]
+
+      # Save tiers into @edit
+      (0..@sb[:num_tiers][detail_index].to_i - 1).each do |tier_index|
+        @edit [:new][:tiers][detail_index][tier_index] = {} unless @edit[:new][:tiers][detail_index][tier_index]
```

Comment:
Is that a space after `@edit`?
```ruby
@edit [:new][:tiers][detail_index][tier_index] = {}
```

Please try to use local variables over these repetitive deeply nested hash accessors.  It's really hard to understand what this code is doing.